var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -602,6 +602,17 @@ def setup_build_environment(self, env):
             env.set('ac_cv_header_gpfs_fcntl_h', 'no')
 
     def configure_args(self):
+        if self.spec.satisfies('+legacylaunchers schedulers=slurm'):
+            # Deleting the links to orterun avoids users running their
+            # applications via mpirun or mpiexec, and leaves srun as the only
+            # sensible choice (orterun is still present, but normal users don't
+            # know about that). See https://github.com/spack/spack/pull/10340
+            # for more details.
+            tty.warn(self.spec.format(
+                "The preferred way to run an application when Slurm is the "
+                "scheduler is to let Slurm manage process spawning via PMI: "
+                "consider building {name} with '~legacylaunchers'"))
```

Comment:
I'm concerned a mere warning here isn't enough to change the behavior -- especially when installing a huge environment, and especially since the change in default could mess with existing scripts.

@alalazo Is it reasonable to have a value variant with  "on/off/auto", where "auto" is the default and it disables legacy launchers when slurm is enabled?

Reply:
a nit here, but potentially a really big one,  drop the "via PMI" in the message.  Starting with the upcoming Open MPI 5 release, srun will only be usuable if the SLURM installation was built with PMIx support. Hence the suggestion to drop "via PMI".  

Also it should be noted that we know some features of Open MPI are not supported via srun, for example MPI_Comm_spawn and company.   There are other limitations when Open MPI's mpirun launcher is not available.  See for example https://github.com/open-mpi/ompi/issues/8299.